lib/mongoid/attributes.rb

```diff
@@ -121,6 +121,7 @@ def read_attribute_before_type_cast(name)
     # @raise [ Errors::ReadonlyAttribute ] If the field cannot be removed due
     #   to being flagged as reaodnly.
     def remove_attribute(name)
+      validate_dots_and_dollars!(name.to_s)
```

Comment:
I would suggest something like `validate_writable_field_name!` to reflect the purpose of the method better rather than its implementation.

Reply:
done